sqlx-postgres/src/type_info.rs

```diff
@@ -1152,7 +1152,10 @@ impl PartialEq<PgType> for PgType {
             true
         } else {
             // Otherwise, perform a match on the name
-            self.name().eq_ignore_ascii_case(other.name())
+            // If either name has no schema qualification, match only on the name part.
+            // If both names have a schema qualification, match both the schema and name parts.
+            std::iter::zip(self.name().rsplitn(2, '.'), other.name().rsplitn(2, '.'))
+                .all(|(part, other_part)| part.eq_ignore_ascii_case(other_part))
```

Comment:
PostgreSQL doesn't use case-insensitive matching for escaped names. `public."TypeName"`, for example, won't match `public.typename` (but `public.TYPENAME`, without escaping, will). I suggest we simply treat escaped and non-escaped names on an entirely separate branch, using case-insensitive ASCII comparison for non-quoted names, and using strict utf-8 equality otherwise.